### PR TITLE
Unit test for groupby.rolling.corr

### LINKED ISSUE #44078: BUG: groupby.rolling.corr incorrect/fails when grouping columns contain tuples
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
df = pd.DataFrame({'a': [(1,), (1,), (1,)], 'b': [4, 5, 6]})
gb = df.groupby(['a'])
print(gb.rolling(2).corr(other=df))

df = pd.DataFrame({'a': [(1,2,), (1,2,), (1,2,)], 'b': [4, 5, 6]})
gb = df.groupby(['a'])
print(gb.rolling(2).corr(other=df))
```

### Issue Description

The first block produces

```
      a    b
a           
1 0 NaN  NaN
  1 NaN  1.0
  2 NaN  1.0
```

where the first index level should be `(1, )` instead. The second block raises a  ValueError. It is due to usage of maybe_make_list in _apply_pairwise (related: #44056). Instead of using this function, we can pack the index keys in lists when `self._grouper.nkeys` is 1 and otherwise leave them as-is. I.e.

```
if self._grouper.nkeys == 1:
    gb_pairs = (
        [pair] for pair in self._grouper.indices.keys()
    )
else:
    gb_pairs = (
        pair for pair in self._grouper.indices.keys()
    )
```

### Expected Behavior

```
          a    b
a           
(1, ) 0 NaN  NaN
      1 NaN  1.0
      2 NaN  1.0

           a    b
a           
(1, 2) 0 NaN  NaN
       1 NaN  1.0
       2 NaN  1.0
```

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by omar-elbaz**
- [ ] closes #44078  
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) 
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by omar-elbaz** [comment]
pre-commit.ci autofix

**Comment by jreback** [comment]
thanks @omar-elbaz

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -3159,3 +3159,29 @@ def test_groupby_series_with_datetimeindex_month_name():
     expected = Series([2, 1], name="jan")
     expected.index.name = "jan"
     tm.assert_series_equal(result, expected)
+
+
+def test_rolling_corr_grouping_column_contains_tuples_1(df):
```
Comment (mroeschke) [inline pandas/tests/groupby/test_groupby.py:3164]: Could you move these tests to `pandas/tests/window/test_groupby.py` in a follow up PR?

Comment (omar-elbaz) [reply pandas/tests/groupby/test_groupby.py:3164]: Yeah sounds good, will get it in shortly

Comment (omar-elbaz) [reply pandas/tests/groupby/test_groupby.py:3164]: Hey @mroeschke, this is the number for the new pr #54504, I'm just waiting for the checks to clear.

Comment (omar-elbaz) [reply pandas/tests/groupby/test_groupby.py:3164]: @mroeschke the other PR went through please merge :) thanks

**Pull request merged**